src/dawn/CodeGen/CXXNaive/CXXNaiveCodeGen.cpp

```diff
@@ -0,0 +1,527 @@
+      stencilFunMethod.addArg("const int i");
+      stencilFunMethod.addArg("const int j");
+      stencilFunMethod.addArg("const int k");
+
+      std::unordered_map<std::string, std::string> paramNameToType;
+      for(std::size_t m = 0; m < fields.size(); ++m) {
+
+        std::string paramName = stencilFun->getOriginalNameFromCallerAccessID(fields[m].AccessID);
+        paramNameToType.emplace(paramName, stencilFnTemplates[m]);
+
+        // each parameter being passed to a stencil function, is wrapped around the ParamWrapper
+        // that contains the storage and the offset, in order to resolve offset passed to the
+        // storage during the function call. For example:
+        // fn_call(v(i+1), v(j-1))
+        stencilFunMethod.addArg("ParamWrapper<" + c_gt() + "data_view<StorageType" +
+                                std::to_string(m) + ">> pw_" + paramName);
+      }
+
+      ASTStencilBody stencilBodyCXXVisitor(stencilInstantiation, paramNameToType,
+                                           StencilContext::E_StencilFunction);
+
+      stencilFunMethod.startBody();
+
+      for(std::size_t m = 0; m < fields.size(); ++m) {
+
+        std::string paramName = stencilFun->getOriginalNameFromCallerAccessID(fields[m].AccessID);
+
+        stencilFunMethod << c_gt() << "data_view<StorageType" + std::to_string(m) + "> "
+                         << paramName << " = pw_" << paramName << ".dview_;";
+        stencilFunMethod << "auto " << paramName << "_offsets = pw_" << paramName << ".offsets_;";
+      }
+
+      stencilBodyCXXVisitor.setCurrentStencilFunction(stencilFun.get());
+      stencilBodyCXXVisitor.setIndent(stencilFunMethod.getIndent());
+      for(const auto& statementAccessesPair : stencilFun->getStatementAccessesPairs()) {
+        statementAccessesPair->getStatement()->ASTStmt->accept(stencilBodyCXXVisitor);
+        stencilFunMethod.indentStatment();
+        stencilFunMethod << stencilBodyCXXVisitor.getCodeAndResetStream();
+      }
+
+      stencilFunMethod.commit();
+    }
+  }
+
+  // generate code for base class of all the inner stencils
+  Structure sbase = StencilWrapperClass.addStruct("sbase", "");
+  MemberFunction sbase_run = sbase.addMemberFunction("virtual void", "run");
+  sbase_run.startBody();
+  sbase_run.commit();
+  sbase.commit();
+
+  // Stencil members:
+  // names of all the inner stencil classes of the stencil wrapper class
+  std::vector<std::string> innerStencilNames(stencils.size());
+  // generate the code for each of the stencils
+  for(std::size_t stencilIdx = 0; stencilIdx < stencils.size(); ++stencilIdx) {
+    const Stencil& stencil = *stencilInstantiation->getStencils()[stencilIdx];
+
+    if(stencil.isEmpty())
+      continue;
+
+    // fields used in the stencil
+    const auto& StencilFields = stencil.getFields();
+
+    innerStencilNames[stencilIdx] = "stencil_" + std::to_string(stencilIdx);
+
+    auto non_temp_fields =
```

Comment:
why not camelCase?

Reply:
auch, I am mixing several projects with differnet conventions.... I think my brain gets blurry sometimes...